spokestack/extensions/webrtc/ans.pyx

```diff
@@ -0,0 +1,59 @@
+"""
+This module contains the AutomaticNoiseSuppression class implemented in Cython
+"""
+# distutils: sources = filter_audio/ns/noise_suppression_x.c
+# distutils: include_dirs = filter_audio/ns/include/
+cimport cans
+cimport numpy as np
+
+
+np.import_array()
+
+cdef class AutomaticNoiseSuppression:
+    cdef cans.NsxHandle *_ans
+    cdef int _sample_rate
+    cdef int _policy
+    cdef int _frame_width
+
+    def __dealloc__(self):
+        cans.WebRtcNsx_Free(self._ans)
+        self._ans = NULL
+
+    def __init__(self, sample_rate=16000, policy=0, **kwargs):
+        self._ans = NULL
+        self._sample_rate = sample_rate
+        self._policy = policy
+        self._frame_width = self._sample_rate * 10 / 1000
+
+        if self._frame_width % 10 != 0:
+            raise ValueError
+
+        result = cans.WebRtcNsx_Create(&self._ans)
+        if result == 0:
```

Comment:
we are silently ignoring failure here and in the check below
failures in these checks should result in exceptions